sklearn/cluster/tests/test_optics.py

```diff
@@ -245,6 +246,40 @@ def test_dbscan_optics_parity(eps, min_samples):
     assert percent_mismatch <= 0.05
 
 
+def test_min_samples_min_clusters_float():
+    # check if OPTIICS is callable with float parameters for
+    # min_samples and min_cluster_size
+    rng = np.random.RandomState(0)
+    n_points_per_cluster = 5
+
+    C1 = [-5, -2] + .8 * rng.randn(n_points_per_cluster, 2)
```

Comment:
any reason not to use a simpler dataset construction, or else factor this out of the tests if it's reused?

Reply:
Just took the setup of another test (from test_extract_xi that qin mentioned earlier).
Is simplification to 3 clusters +1 noise okay?